gammapy/modeling/models/spatial.py

```diff
@@ -1003,10 +1010,16 @@ def __init__(
         self, map, meta=None, normalize=True, interp_kwargs=None, filename=None,
     ):
         if (map.data < 0).any():
-            log.warning("Diffuse map has negative values. Check and fix this!")
+            log.warning("Map has negative values. Check and fix this!")
 
         if filename is not None:
             filename = str(make_path(filename))
+            if not os.path.isfile(filename):
+                raise IOError("File not found")
+        else:
+            log.warning(
```

Comment:
Again I think having this check in `__init__` doesn't make sense to me. The model could be created without being serialise...why not move to `.to_dict()` instead?

Reply:
moved to `to_dict()`